azext_edge/edge/commands_namespaces.py

```diff
@@ -917,3 +918,549 @@ def query_namespace_assets(
+        asset_type="custom",
+        dataset_data_source=dataset_data_source,
+        datasets_custom_configuration=dataset_configuration,
+        dataset_destinations=dataset_destinations,
+        replace=replace,
+        **kwargs
+    )
+
+
+def add_namespace_opcua_asset_dataset(
+    cmd,
+    asset_name: str,
+    namespace_name: str,
+    resource_group_name: str,
+    dataset_name: str,
+    dataset_data_source: str,
+    dataset_destinations: Optional[str] = None,
+    opcua_dataset_publishing_interval: Optional[int] = None,
+    opcua_dataset_sampling_interval: Optional[int] = None,
+    opcua_dataset_queue_size: Optional[int] = None,
+    opcua_dataset_key_frame_count: Optional[int] = None,
+    opcua_dataset_start_instance: Optional[str] = None,
+    replace: Optional[bool] = False,
+    **kwargs
+) -> dict:
+    return NamespaceAssets(cmd).add_dataset(
+        asset_name=asset_name,
+        namespace_name=namespace_name,
+        resource_group_name=resource_group_name,
+        dataset_name=dataset_name,
+        asset_type=DeviceEndpointType.OPCUA.value,
+        dataset_data_source=dataset_data_source,
+        dataset_destinations=dataset_destinations,
+        opcua_dataset_publishing_interval=opcua_dataset_publishing_interval,
+        opcua_dataset_sampling_interval=opcua_dataset_sampling_interval,
+        opcua_dataset_queue_size=opcua_dataset_queue_size,
+        opcua_dataset_key_frame_count=opcua_dataset_key_frame_count,
+        opcua_dataset_start_instance=opcua_dataset_start_instance,
+        replace=replace,
+        **kwargs
+    )
+
+
+def list_namespace_asset_datasets(
+    cmd,
+    asset_name: str,
+    namespace_name: str,
+    resource_group_name: str
+) -> List[dict]:
+    return NamespaceAssets(cmd).list_datasets(
+        asset_name=asset_name,
+        namespace_name=namespace_name,
+        resource_group_name=resource_group_name
+    )
+
+
+def show_namespace_asset_dataset(
+    cmd,
+    asset_name: str,
+    namespace_name: str,
+    resource_group_name: str,
+    dataset_name: str
+) -> dict:
+    return NamespaceAssets(cmd).show_dataset(
+        asset_name=asset_name,
+        namespace_name=namespace_name,
+        resource_group_name=resource_group_name,
+        dataset_name=dataset_name
+    )
+
+
+def update_namespace_custom_asset_dataset(
+    cmd,
+    asset_name: str,
+    namespace_name: str,
+    resource_group_name: str,
+    dataset_name: str,
+    dataset_data_source: Optional[str] = None,
+    dataset_configuration: Optional[str] = None,
+    dataset_destinations: Optional[str] = None,
+    **kwargs
+) -> dict:
+    return NamespaceAssets(cmd).update_dataset(
+        asset_name=asset_name,
+        namespace_name=namespace_name,
+        resource_group_name=resource_group_name,
+        dataset_name=dataset_name,
+        asset_type="custom",
+        dataset_data_source=dataset_data_source,
+        datasets_custom_configuration=dataset_configuration,
```

Comment:
the `abc=abc` pattern a bit annoying - I get that it's useful for readability and handling optional parameters, but it feels verbose and sometimes parameter name doesn't match with each other on both side 🌵

Reply:
ok going to try to go through and fix stuff up